src/Notifier.js

```diff
@@ -16,8 +16,10 @@ limitations under the License.
 
```

Comment:
The flexibility is provided through the component system, nothing more. We allow people to either clobber specific components or write their own, and we can design reusable components which allow them to use the same code that we use in Vector. The exposed API surface is simple and understandable to an outsider (it's a dictionary of `string => Component`, and is accessed via `sdk.getComponent("string")`). I like this simplicity. I don't think we need to add anything to this.

I think it is a mistake to add more layers of complexity by adding more systems to make "flexible" logic classes (e.g. A React-classless `Notifier` which looks and smells like a React class but isn't one, a `Modulator` system which is used precisely once). Logic classes are a lot easier to dependency inject (e.g. through constructor injection) than a tree of UI components, which is why `getComponent()` makes a lot of sense for UI components but makes less sense for logic classes IMO.